AAT_gae/src/main/java/com/aat/restlet/AttendanceResource.java

```diff
@@ -19,17 +19,24 @@
 import com.aat.datastore.Course;
 import com.aat.utils.ResourceUtil;
 
+import java.util.List;
+
 public class AttendanceResource extends ServerResource {
 	
 	@Post
 	public Representation create(Representation entity) {
+		// Check if of type Student
+		ResourceUtil.checkTokenPermissions(this, Student.class);
```

Comment:
This method is called from all methods of CourseResource, AttendenceResource, GroupResource, which do some modification in datastore (post, put and delete methods).
Resources which do not call this method are ones retrieving list of objects (GroupsResource, CoursesResource) and also Signin and Signup resources.